### PR TITLE
Added asy source for rough version of diagram.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Rough outline of the figure with sites and mutations. I have a few ideas for making it better:

1. Add a time scale for the trees
2. Add mutational states for all the samples on the bottom (underneath ID label)
3. Possibly add state transitions to the mutation labels (i.e. C->G)
4. Maybe reclaim some x-space by  mutations table underneath the sites table. (Not certain this will look good though, and might end up confusing).

Also short discussion of sites and mutations.

### CODE REVIEW

**Comment by petrelharp** [comment]
Looks good.  If the diagram contains all the information in the tables, that'd make it a very useful illustration.  I'm in favor of all proposed ideas 1--4.

**Comment by jeromekelleher** [comment]
Cool. I'll update the diagram and make a pass at the rest of the text for this section tomorrow.

**Comment by jeromekelleher** [comment]
I've added some more discussion here based around an example simulation in msprime. I think this is an OK-ish first pass at this section, so we might as well merge this much. I'll make another pass at the diagram and text after going through some other stuff.

Any thoughts @petrelharp?

File: forwards_paper.tex [hunk 1]
```diff
@@ -215,31 +215,47 @@ \subsection*{Tree sequences}
...
 mutation resulting in node $2$ inheriting the state 'T'. The second side (in the
-right hand tree) has two mutations: one occuring over node $3$ changing the state to
+right hand tree) has two mutations: one occurring over node $3$ changing the state to
 `C' and a back mutation over node $1$ changing the state to `G'.
 \jk{Not sure if I'm labouring the point here: this should be reasonably obvious.
 Perhaps if we added the state changes to the trees, and the states of the leaves
 below the samples, we could get rid of all the example discussion here?}
 
-
-\jk{Other stuff we can review here:}
```
Comment (petrelharp) [inline forwards_paper.tex]: I'm in favor of reviewing any/all of these things.   Probably can't emphasize this too much.

File: forwards_paper.tex [hunk 2]
```diff
@@ -215,31 +215,47 @@ \subsection*{Tree sequences}
...
+This encoding of a sequence of trees and accompanying mutational information is
+very concise. To illustrate this, we ran a simulation of $500,000$ samples of a
+$200$ megabase human-like chromosome ($N_e=10^4$ and per-base mutation and
+recombination rates of $10^{-8}$ per generation) using \msprime. This resulted
+in about 1 million distinct marginal trees and $1.1$ million infinite-sites
+mutations. The HDF5 file encoding the nodes, edges, sites and mutations (as
+described above) for this simulation consumed 157MiB of storage space. Using
+the \msprime\ Python API, the time required to load this file into memory was
+around 1.5 seconds, and the time required to iterate over all 1 million tree
+was 2.7 seconds. In contrast, recording the topological information in Newick
```
Comment (petrelharp) [inline forwards_paper.tex:233]: Should we see if the journal accepts emojis? :excited_face:  Seriously, this is great to emphasize here.

File: forwards_paper.tex [hunk 3]
```diff
@@ -215,31 +215,47 @@ \subsection*{Tree sequences}
...
+format would require around 20 TiB and storing the genotype information
+in VCF would require about 1 TiB. Working with either the Newick or VCF encoding
+of this dataset would be exceedingly cumbersome, likely requiring several
+days of CPU time simply to read the information info memory. \jk{Could
+add concrete timing estimates based on subsets here, but maybe overdoing it?}
```
Comment (petrelharp) [inline forwards_paper.tex:238]: nah, this seems good

File: forwards_paper.tex [hunk 4]
```diff
@@ -215,31 +215,47 @@ \subsection*{Tree sequences}
...
+from observed sequence data requires $O(n^2 m)$ time, since we must
+compare all pairs of samples at every site. However, given the topology of
+the trees at every site and the locations of mutations much better algorithms
+are possible. Using the fact that we can count the number of samples below
+a given node efficiently using previously described tree sequence
+algorithms~\citep{kelleher2016efficient}, the time required to compute $\pi$
+becomes roughly $O(n \log n + m)$.
+The \msprime\ API provides a method to compute $\pi$ among arbitrary subsets of the
+samples in a tree sequence, and calculating $\pi$ over all 500K samples
+in the example above required about 1.2 seconds.
```
Comment (petrelharp) [inline forwards_paper.tex:254]: also a good example.  I agree a comparison would be good.

**Comment by petrelharp** [comment]
great. merging.

File: forwards_paper.tex (hunk 4 above)
Comment (jeromekelleher) [reply forwards_paper.tex:254]: Cool. I'll see if I can get pylibseq to do the trick so.

**Pull request merged**